core/templates/pages/exploration-player-page/exploration-player-page.component.ts

```diff
@@ -33,17 +35,39 @@ require(
 
 require('interactions/interactionsRequires.ts');
 require('objects/objectComponentsRequiresForPlayers.ts');
+require('pages/exploration-player-page/services/command-executor.service.ts');
 
 angular.module('oppia').component('explorationPlayerPage', {
   template: require('./exploration-player-page.component.html'),
   controller: [
     'ContextService', '$timeout', 'PageTitleService',
     'ReadOnlyExplorationBackendApiService',
+    'CommandExecutorService', 'WindowWrapperMessageService',
     function(
         ContextService, $timeout, PageTitleService,
-        ReadOnlyExplorationBackendApiService) {
+        ReadOnlyExplorationBackendApiService,
+        CommandExecutorService, WindowWrapperMessageService) {
       var ctrl = this;
+      var keyword = 'secret_hostname';
+      var numberOfSetEntries = 0;
+      var getUrlParams = function() {
+        var urlParams = {};
+        var parts = WindowWrapperMessageService.getLocationHref().replace(
+          /[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
+            urlParams[key] = value;
+          });
+        return urlParams;
+      };
       ctrl.$onInit = function() {
+        var urlParams = getUrlParams();
+
+        if (urlParams[keyword] !== undefined &&
+          ServicesConstants.WHITELISTED_IFRAME_SECRETS.indexOf(
+            urlParams[keyword]) >= 0) {
+          CommandExecutorService.initialize();
+          CommandExecutorService.sendParentReadyState();
```

Comment:
Is there a reason why sendParentReadyState() isn't part of initialize()?

In fact I'm not sure you need this at all here; it doesn't really have anything to do with the exploration player, and we don't want to repeat this code from other places where CommandExecutorService is invoked. Couldn't CommandExecutorService take care of its own initialization?